app/src/main/java/couclou/fr/nixconfighelper/MainAdapter.kt

```diff
@@ -25,24 +39,126 @@ class MainAdapter(val feed:Feed): RecyclerView.Adapter<CustomViewHolder>() {
             val readOnly = element.readOnly
             val defaultValue = element.defaultValue
             var description = element.description.replace("\n","")
-
-            holder.view.keynameValue.text = keyname
-            holder.view.typeValue.text = type
-            holder.view.readOnlyValue.text = readOnly.toString()
-            holder.view.defaultValueValue.text = defaultValue
-            holder.view.descriptionValue.text = description
-
             var declarations: String = ""
-            if (declarations.isNotEmpty()) {
+            if (element.declarations.isNotEmpty()) {
                 for (declaration in element.declarations) {
-                    declarations = "$declarations \n $declaration"
+                    declarations = "$declarations\n$declaration"
```

Comment:
Tu as de beaucoup plus jolies fonctions pour faire ça : https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to-string.html

Implémenter ça comme ça, c'est de la complexité en trop.